distro/wildfly/subsystem/src/main/java/org/camunda/bpm/container/impl/jboss/service/MscManagedProcessEngineController.java

```diff
@@ -236,52 +247,39 @@ protected JakartaTransactionProcessEngineConfiguration createProcessEngineConfig
   private Object createInstance(String configurationClassName) {
     try {
       Class<?> configurationClass = getClass().getClassLoader().loadClass(configurationClassName);
-      return configurationClass.newInstance();
+      return configurationClass.getDeclaredConstructor().newInstance();
 
     } catch (Exception e) {
       throw new ProcessEngineException("Could not load '"+configurationClassName+"': the class must be visible from the camunda-wildfly-subsystem module.", e);
     }
   }
 
-  public Injector<TransactionManager> getTransactionManagerInjector() {
-    return transactionManagerInjector;
-  }
-
-  public Injector<DataSourceReferenceFactoryService> getDatasourceBinderServiceInjector() {
-    return datasourceBinderServiceInjector;
-  }
-
-  public InjectedValue<MscRuntimeContainerJobExecutor> getMscRuntimeContainerJobExecutorInjector() {
-    return mscRuntimeContainerJobExecutorInjector;
-  }
-
-  public static void initializeServiceBuilder(ManagedProcessEngineMetadata processEngineConfiguration, MscManagedProcessEngineController service,
-    ServiceBuilder<ProcessEngine> serviceBuilder, String jobExecutorName) {
+  public void initializeServiceBuilder(ManagedProcessEngineMetadata processEngineConfiguration,
+      ServiceBuilder<?> serviceBuilder, ServiceName name, String jobExecutorName) {
 
     ContextNames.BindInfo datasourceBindInfo = ContextNames.bindInfoFor(processEngineConfiguration.getDatasourceJndiName());
-    serviceBuilder.addDependency(ServiceName.JBOSS.append("txn").append("TransactionManager"), TransactionManager.class, service.getTransactionManagerInjector())
-      .addDependency(datasourceBindInfo.getBinderServiceName(), DataSourceReferenceFactoryService.class, service.getDatasourceBinderServiceInjector())
-      .addDependency(ServiceNames.forMscRuntimeContainerDelegate(), MscRuntimeContainerDelegate.class, service.getRuntimeContainerDelegateInjector())
-      .addDependency(ServiceNames.forMscRuntimeContainerJobExecutorService(jobExecutorName), MscRuntimeContainerJobExecutor.class, service.getMscRuntimeContainerJobExecutorInjector())
-      .setInitialMode(Mode.ACTIVE);
+    transactionManagerSupplier = serviceBuilder.requires(ServiceName.JBOSS.append("txn").append("TransactionManager"));
+    datasourceBinderServiceSupplier = serviceBuilder.requires(datasourceBindInfo.getBinderServiceName());
+    runtimeContainerDelegateSupplier = serviceBuilder.requires(ServiceNames.forMscRuntimeContainerDelegate());
+    mscRuntimeContainerJobExecutorSupplier = serviceBuilder.requires(ServiceNames.forMscRuntimeContainerJobExecutorService(jobExecutorName));
+    serviceBuilder.setInitialMode(Mode.ACTIVE);
     serviceBuilder.requires(ServiceNames.forMscExecutorService());
 
     if(processEngineConfiguration.isDefault()) {
       serviceBuilder.addAliases(ServiceNames.forDefaultProcessEngine());
```

Comment:
❓ `ServiceBuilder#addAliases` is also deprecated. Can we resolve it?

Reply:
Yes.